### PR TITLE
sigh: clean obsolete build outputs; tidy up findProjectFiles usage

### PR DESCRIPTION
**Pull request by mykmartin**

### CODE REVIEW

File: tools/sigh.ts [hunk 1]
```diff
@@ -318,17 +323,37 @@ function railroad(): boolean {
   return true;
 }
 
-function buildPath(path: string, deps?: string[]): () => boolean {
+function cleanObsolete() {
```
Comment (csilvestrini) [inline tools/sigh.ts]: A comment for this function would be nice

File: tools/sigh.ts [hunk 2]
```diff
@@ -318,17 +323,37 @@ function railroad(): boolean {
...
+  const exts = ['js', 'js.map', 'd.ts'];
+  for (const file of [...findProjectFiles('build', /javaharness|sigh\.js/, /\.js$/)]) {
```
Comment (csilvestrini) [inline tools/sigh.ts]: I don't think you need to turn this into an array, you can just iterate over it directly

File: tools/sigh.ts [hunk 3]
```diff
@@ -318,17 +323,37 @@ function railroad(): boolean {
...
+    const buildBase = file.slice(0, -2);
```
Comment (csilvestrini) [inline tools/sigh.ts]: Add comment: // drop js file extension

File: tools/sigh.ts (hunk 1 above)
Comment (mykmartin) [reply tools/sigh.ts]: Done.

File: tools/sigh.ts (hunk 2 above)
Comment (mykmartin) [reply tools/sigh.ts]: I do, actually - the loop removes files from the disk which causes problems in the readdir call.

File: tools/sigh.ts (hunk 3 above)
Comment (mykmartin) [reply tools/sigh.ts]: Done.

**Pull request merged**